demo/sensitivity.R

```diff
@@ -1,141 +1,159 @@
-##  (frequency/ratio of days of flow below 20%ile)
-##  to IHACRES-CWI model parameters using a subset of data from Cotter catchment
-
-## Calculate observed 20%ile flow
-thres.Q20 <- as.numeric(quantile(obs$Q, probs = c(0.2), na.rm = TRUE))
-## Define function to calculate number of days below thres.Q20
-F20 <- function(X) length(which(X < thres.Q20)) / length(X)
-
-mm <- morris(
-  model = evalPars,
-  factors = names(getFreeParsRanges(modx)),
-  r = 4,
-  design = list(
-    type = "oat",
-    levels = 10,
-    grid.jump = 2
-  ),
-  binf = sapply(getFreeParsRanges(modx), min),
-  bsup = sapply(getFreeParsRanges(modx), max),
-  object = modx,
-  ## Change objective to use the prediction function defined above
-  ## See ?objFunVal
-  ## The objective function can refer to Q and X,
-  ##  representing observed and modelled flow, respectively.
-  ##  It should return a single numeric value.
-  objective = ~ F20(X)
-)
-
-print(mm)
-plot(mm, main = "Sensitivity F20~IHACRES-CWI parameters with Cotter data")
-
-## See the Morris (1991) journal paper(?morris) for a more comprehensive interpretation.
-## These results are only an example, but if sampling were sufficient, the results would
-##  suggest that NSE* is insensitive to tau_s (mu.star=0.065), but F20 is sensitive to
-##  tau_s (mu.star=0.34)
-## The parameter value selected therefore affects the prediction, so tau_s cannot be
-##  fixed to improve identifiability unless its value is sufficiently certain.
-## Using a different objective function, e.g. NSElog* might improve identifiability.
-## The settings for Morris used may also influence results. Try changing: the parameter ranges,
-##  number of replicates, number of levels and length of data series.
-## A SOBOL sensitivity analysis would also provide confidence intervals around the TSI.
-
-
-################################################################################
-## Sensitivity using SOBOL2002 method of NSElog* prediction function to
-##  IHACRES-CWI model parameters using a subset of data from Cotter catchment
-## This might take a while, potentially ~15min
-
-## Save current time
-st <- proc.time()
-
-## Set number of samples desired
-n <- 1000
-ss <- sobol2002(
-  model = evalPars,
-  ## Draw two random samples of parameters
-  X1 = parameterSets(getFreeParsRanges(modx), n),
-  X2 = parameterSets(getFreeParsRanges(modx), n),
-  ## Number of bootstrap replicates
-  nboot = 100,
-  ## Arguments to be passed to evalPars
-  object = modx,
-  ## NSElog* objective function (using the logarithm of Q and X)
-  objective = ~ hmadstat("r.sq.log")(Q, X) / (2 - hmadstat("r.sq.log")(Q, X))
-)
-
-## Print elapsed time
-print(proc.time() - st)
-
-## Show results
-print(ss)
-plot(ss)
-
-## tau_s appears to still be insensitive with NSElog* (TSI=0.127), suggesting
-## the information in this dataset may not be sufficient to identify this parameter.
+#' Support for sensitivity analysis in hydromad
```

Comment:
This is in the demo folder. Is it correct that Roxygen supports extracting annotations from demos into Rd files?
IIRC demos also show comments to the user when they are called. Does this hold true for Roxygen comments?

Reply:
To my knowledge, roxygen2 only looks at the `R` folder to generate .Rd files.

I think what happened here was that Rd2roxygen found a file called hydromad_sensitivity.Rd (present in the current `master` [here](https://github.com/josephguillaume/hydromad/blob/master/man/hydromad_sensitivity.Rd)) in `man` and converted it to a roxygen2 tagged .R file which was placed into `R`. I must have later moved the .R file to the `demo` folder to preserve the overall folder structure. However, since the .R file was moved out of `R`, this means that hydromad_sensitivity.Rd was deleted when `devtools::document` or 
`roxygenise` were used. 

If you want a roxygen2 generated .Rd file of this, we may need to have a copy of sensitivity.R in `R`, and a version sans the annotations in `demo`.

Reply:
I definitely want to keep a man file for this and it sounds like we shouldn't mix Rd files with roxygen2, so it would be great if you could move the Roxygen2 annotation back to R and revert the demo. 
Thanks!

Reply:
In attempting to fix this, I found out that `roxygen2::roxygenise` won't generate a .Rd file unless there is an actual function in there being documented. However, it does seem like you can have documentation that roxygen2 hasn't generated been generated still in `man` that isn't deleted when `devtools::document` or `roxygenise` are run ... so I will pop this into the next push.